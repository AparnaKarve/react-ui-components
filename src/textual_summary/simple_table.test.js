import React from 'react';
import renderer from 'react-test-renderer';
import SimpleTable from './simple_table';

describe('Simple Table', () => {
  it('renders just fine...', () => {
    const tableData = {
      labels: ['Network Protocol', 'Host Protocol', 'Direction', 'Port', 'End Port', 'Source'],
      rows: [
        [null, '-1', 'outbound', 0, 0, '0.0.0.0/0'],
        [null, 'TCP', 'inbound', 22, 22, '0.0.0.0/0'],
        [null, 'TCP', 'inbound', 80, 80, '0.0.0.0/0'],
      ],
      title: 'Firewall Rules',
    };

    /*
     * label:
     *   a) simple
     *   b) object {sortable: true} TODO
     * rows
     *   value:
     *     a) simple
     *     b) object {expandable: true} TODO
     */
    const component = renderer.create(<SimpleTable
      title={tableData.title}
      labels={tableData.labels}
      rows={tableData.rows}
    />);
    const table = component.toJSON();
    expect(table).toMatchSnapshot();
  });
});
