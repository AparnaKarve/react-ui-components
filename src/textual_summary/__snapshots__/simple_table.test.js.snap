// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Table renders just fine... 1`] = `
<table
  className="table table-bordered table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan={6}
      >
        Firewall Rules
      </th>
    </tr>
    <tr>
      <td>
        <strong>
          Network Protocol
        </strong>
      </td>
      <td>
        <strong>
          Host Protocol
        </strong>
      </td>
      <td>
        <strong>
          Direction
        </strong>
      </td>
      <td>
        <strong>
          Port
        </strong>
      </td>
      <td>
        <strong>
          End Port
        </strong>
      </td>
      <td>
        <strong>
          Source
        </strong>
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td />
      <td>
        -1
      </td>
      <td>
        outbound
      </td>
      <td>
        0
      </td>
      <td>
        0
      </td>
      <td>
        0.0.0.0/0
      </td>
    </tr>
    <tr>
      <td />
      <td>
        TCP
      </td>
      <td>
        inbound
      </td>
      <td>
        22
      </td>
      <td>
        22
      </td>
      <td>
        0.0.0.0/0
      </td>
    </tr>
    <tr>
      <td />
      <td>
        TCP
      </td>
      <td>
        inbound
      </td>
      <td>
        80
      </td>
      <td>
        80
      </td>
      <td>
        0.0.0.0/0
      </td>
    </tr>
  </tbody>
</table>
`;
